sql/src/main/java/io/crate/analyze/PrivilegesAnalyzer.java

```diff
@@ -95,22 +101,35 @@ private static void ensureUserManagementEnabled(@Nullable User user) {
         return privilegeTypes;
     }
 
-    private void validateTableNames(List<QualifiedName> tableNames) {
-        tableNames.forEach(t -> schemas.getTableInfo(TableIdent.fromIndexName(t.toString())));
+    private void validateTableNames(List<String> tableNames) {
+        tableNames.forEach(t -> {
+            TableIdent ident = TableIdent.fromIndexName(t);
+            List<String> schemaInfo = Collections.singletonList(ident.schema());
+            validateSchemaNames(schemaInfo);
+            schemas.getTableInfo(ident);
+        });
     }
 
-    private void validateSchemaNames(List<QualifiedName> schemaNames) {
-        schemaNames.forEach(s ->
-            schemas.validateSchemaName(s.toString())
+    private void validateSchemaNames(List<String> schemaNames) {
+        String message = "GRANT/DENY/REVOKE Privileges on information_schema is not supported";
+        schemaNames.forEach(s -> {
+                if (InformationSchemaInfo.NAME.equals(s)) {
```

Comment:
can you pls factor this out into a method and use it also at `validateTableNames` to avoid the singletonList creation?